aip/general/0215.md

```diff
@@ -7,59 +7,57 @@ placement:
-do not actually support it. This is surprising to users, and creates a
-situation where the proto files are not accurate representations of the API
-surface.
-
-Additionally, client library generators usually generate a class for each
-message in your protos in a language-appropriate namespace. In several
-languages, each version of your API is shipped as a _separate_ client library,
-and the code generator needs to generate the common messages for your API in
-order to ensure a complete package. This makes it difficult to use multiple
-APIs with similar dependent messages together.
-
-For omitted versions, the version is effectively hidden, and it becomes more
-difficult to reason about release phases (alpha, beta, GA), with no substantive
-benefit.
+APIs are mostly defined in terms of protos which are API-specific, with occasional
+dependencies on common protos. Keeping APIs isolated from each other avoids versioning
+problems and client library packaging problems.
 
 ## Guidance
 
-- All protos specific to an API **should** be within the versioned package
-  (e.g., `yourapi.v1.SharedProtoMessage`).
-
-- In scenarios where an API doesn't consider itself to have a version, the API
-  **must** use `v1`. (Omitted-version protos are prohibited.)
-
-- When a shared proto is identical, that proto **should** be duplicated to the
-  other versioned package (e.g., copied and pasted into
-  `yourapi.v2.SharedProtoMessage`).
-
-### What if a proto will never change?
-
-There are some situations where it is useful to have an unversioned
-proto. These should generally apply to a complete organization or suite
-of APIs, not just a single API. In these situations, it may be possible
-to add it to our collections of common protos. Common protos are able to
-be used by many APIs, and are always unversioned.
-
-Common protos shall always have a package structure ending in `type` (e.g.
-`google.type` or `google.cloud.type`).
-
-**Warning:** This is a relatively rare occurrence. If you find yourself
-adding protos to your organizations's `type` directory frequently,
-double-check that this is actually where they belong.
-
-For more information on common protos, consult [AIP-213][].
+- All protos specific to an API **must** be within a package with a major version
+  (e.g., `google.library.v1`).
+- References to resources in other APIs **must** be expressed in terms of
+  resource names ([AIP-122][]), rather than using the resource messages.
+- When two versions of an API use effectively the same (API-specific) proto
+  that proto **must** be duplicated in each version. (In other words, APIs
+  **must not** create their own "API-specific common proto" package.)
+- Organization-specific common protos **may** be placed in a common package,
+  as described in [AIP-213][], but **must not** be used by any API outside
+  that organization.
+- Global common protos (also described in AIP-213) **may** be freely used by any API.
```

Comment:
AIP-213 talks about "common components" while the phrase used repeatedly in this file is "common protos".  I don't think it matters which you use for Google, as long as it's consistent.

Of course, for the agnostic aip.dev we'll keep saying "common components".  Maybe this is a weak argument in favor of using "common components" here too, so that the correspondence is clear?

Reply:
The way I've written it is *probably* geared around "protos are files, components are proto elements like messages, enums". That's definitely how I *think* about it anyway. But I strongly suspect that I can rewrite most of this using just components instead of protos - so I'll give that a try.

Reply:
Thanks - FWIW, when I hear someone talk about a "proto" I assume that refers to a message definition, not a `.proto` file.